From df0d5abcc91b95c8f7e496fb71ec44b1b18e6e4d Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <cgharzuzi@habana.ai>
Date: Tue, 28 Jun 2022 13:43:35 +0300
Subject: [PATCH] [SW-85391] greco: clear eeprom ready indication on spsram

U-boot and ARCPID poll on spsram address which indicates
that eeprom data is ready (copied there by arc mgmt), But since
arcmgmt can boot after uboot/arcpid and the spsram data is only
cleared in cold reset it can cause a scenario where arcpid
and uboot read old eeprom data.

Clearing the ready indication in preboot will prevent
the scenario above and will keep arcpid and uboot polling
for refreshed eeprom data.

Change-Id: I493e56a5c536d20650086de403ec50ec52feefb5
Signed-off-by: Christine Gharzuzi <cgharzuzi@habana.ai>
---

diff --git a/arch/greco/arc/common.c b/arch/greco/arc/common.c
index a095e5c..56d61ac 100644
--- a/arch/greco/arc/common.c
+++ b/arch/greco/arc/common.c
@@ -25,6 +25,7 @@
 #include <psoc_arc1_cfg_masks.h>
 
 #include <hl_boot_if.h>				/* requires stdint.h */
+#include <hl_spsram_if.h>
 #include <greco/greco_reg_map.h>
 #include <goya2/goya2.h>
 #include <common/pci_ids.h>
@@ -173,6 +174,11 @@
 {
 	u32 mem_cfg, cpy_sts, rst_src;
 	u64 start;
+	u32 address = SP_SRAM_BASE + EEPROM_SPM_COPY_OFFSET;
+	struct eeprom_spsram_layout *eeprom_copy = (struct eeprom_spsram_layout *)address;
+
+	/* Clear indication on eeprom data ready */
+	*((volatile u32 *)(&eeprom_copy->status)) = 0x0;
 
 	rst_src = hl_rst_src_get();
 	if (rst_src != HL_COLD_RST)
